### PR TITLE
fix(Popover): Switch Popover to use allowOutsideClick

### PR DESCRIPTION
**Pull request by hmafzal**
### Overview

Switches the Popover from using `clickOutsideDeactivates` to using `allowOutsideClick`, to first check if the click is not inside the target element before deactivating.

Related Monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/20497

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@6.0.2-alpha.fa53aa.0<br/>@codecademy/gamut@17.4.4-alpha.fa53aa.0<br/>@codecademy/styleguide@19.0.2-alpha.fa53aa.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fc54cbc92a6fe37133e91e9--gamut-preview.netlify.app>https://5fc54cbc92a6fe37133e91e9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fc54cbc92a6fe37133e91e9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@6.0.2-alpha.bd2b37.0<br/>@codecademy/gamut@17.4.4-alpha.bd2b37.0<br/>@codecademy/styleguide@19.0.2-alpha.bd2b37.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fc552d892ef2b013cee3816--gamut-preview.netlify.app>https://5fc552d892ef2b013cee3816--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fc552d892ef2b013cee3816>Deploy Logs</a></p>

File: packages/gamut-labs/package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "react-dom": ">=16.8.1"
   },
   "dependencies": {
-    "@codecademy/gamut": "^17.4.3",
+    "@codecademy/gamut": "^17.4.4-alpha.fa53aa.0",
```
Comment (hmafzal) [inline packages/gamut-labs/package.json:25]: When updating the gamut dependency here, do i need to push the gamut changes separately? This should be 17.4.4 but that won't be deployed until after this change is merged, right?

**Review by katiezutter — APPROVED** [review]
LGTM, pending the question for web plat on the gamut dependency!

File: packages/gamut-labs/package.json (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut-labs/package.json:25]: Err, I've never had to change package files manually in PRs.

**Review by JoshuaKGoldberg — APPROVED** [review]
Nice. Glad you figured this out 😄

File: packages/gamut-labs/package.json (hunk 1 above)
Comment (hmafzal) [reply packages/gamut-labs/package.json:25]: @JoshuaKGoldberg hmm, I needed to do this to make sure gamut-labs used the right version of gamut when using the new code in the monolith. I guess I can revert this now and it'll automatically update to the newest version once merged?

Comment (JoshuaKGoldberg) [reply packages/gamut-labs/package.json:25]: Yeah - when it's published it'll be bumped.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@6.0.3-alpha.c74c43.0<br/>@codecademy/gamut@17.4.5-alpha.c74c43.0<br/>@codecademy/styleguide@19.0.3-alpha.c74c43.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@6.0.3-alpha.2fd8f9.0<br/>@codecademy/gamut@17.4.5-alpha.2fd8f9.0<br/>@codecademy/styleguide@19.0.3-alpha.2fd8f9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fc65fc0b427190091d26ba3--gamut-preview.netlify.app>https://5fc65fc0b427190091d26ba3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fc65fc0b427190091d26ba3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fc65ff10a3fdd0096a17498--gamut-preview.netlify.app>https://5fc65ff10a3fdd0096a17498--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fc65ff10a3fdd0096a17498>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fc66dc219430b013af1c271--gamut-preview.netlify.app>https://5fc66dc219430b013af1c271--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fc66dc219430b013af1c271>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.0.1-alpha.1d3e21.0<br/>@codecademy/gamut@19.0.1-alpha.1d3e21.0<br/>@codecademy/styleguide@21.0.1-alpha.1d3e21.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fc69ae64b9e2736b4d30b57--gamut-preview.netlify.app>https://5fc69ae64b9e2736b4d30b57--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fc69ae64b9e2736b4d30b57>Deploy Logs</a></p>

**Pull request merged**